frontend/src/components/Modal/Modal.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState, useRef } from "react";
+import "./Modal.scss";
+import Button from "../Button/Button";
+
+const Modal = ({ header, body, footer }) => {
+  const [modalShow, setModalShow] = useState(true);
```

Comment:
the modal should not know if it is visible or not, that state should be managed within it's parent component, just remove that and where ever you call setModalShow you instead call mentioned prop `onClose`